cmake/Export.hh.in

```diff
@@ -52,12 +52,46 @@
 #endif
 
 
-#ifndef IGN_DEPRECATED
+#ifndef GZ_DEPRECATED
 /// For @lib_name@ developers: Use this macro to indicate that a
 /// function or class has been deprecated and should no longer be used. A
 /// version should be specified to provide context to the user about when the
 /// function became deprecated.
-#define IGN_DEPRECATED(version) IGN_DEPRECATED_ALL_VERSIONS
+
+// NOTE(CH3): GZ_DEPRECATED_ALL_VERSIONS can't be ticktocked, it's generated
+// On the bright side, it seems to be private
+#define GZ_DEPRECATED(version) GZ_DEPRECATED_ALL_VERSIONS
 #endif
 
+
+// TICKTOCK IGNITION ========================================================
+// TODO(CH3): Remove on ticktock, supports defining IGNITION macros if we're
+//            a lib is using GZ_ prefixed export macros
+#ifndef @_ign_export_base@_VISIBLE
+/// For @lib_name@ developers: Apply this macro to @lib_name@
+/// functions and classes which consumers of this library will need to be able
+/// to call from their own programs or libraries.
+#define @_ign_export_base@_VISIBLE \
+  DETAIL_@export_base@_VISIBLE
 #endif
+
+
+#ifndef @_ign_export_base@_HIDDEN
+/// For @lib_name@ developers: Apply this macro to @lib_name@
+/// functions and classes which must not be used by consumers of this library.
+/// By default, this property is applied to all classes and functions which are
+/// not tagged with @_ign_export_base@_VISIBLE, so this does not
+/// generally need to be used.
+#define @_ign_export_base@_HIDDEN \
+  DETAIL_@export_base@_HIDDEN
+#endif
+
+
+#ifndef IGN_DEPRECATED
+// TODO(CH3): Remove on ticktock
+// NOTE(CH3): This has no docstring since we'll make GZ_DEPRECATED the default
+#define IGN_DEPRECATED(version) GZ_DEPRECATED_ALL_VERSIONS
```

Comment:
do we want to define `IGNITION_` versions of `_VISIBLE` and `_HIDDEN` as well?

if so, we should create an `ign_export_base` variable to use when configuring this file

Reply:
They're defined in here already, using the `_ign_export_base` variable!

Reply:
Created and populated here: https://github.com/ignitionrobotics/ign-cmake/pull/249/files#diff-955d1f97e2360702839ba9277cc71b9797e65d417d5cb88f383fec0e53ad5124R1447 right above the configure call

Reply:
thanks; I could have done a better job reading the pull request changes before making comments like that. I was a little confused on first glance to see all the Doxygen for the redirection macros; I think we could remove them. See my suggested changes in ee17946737632fec73d1094c111e2fcf793cb908

Reply:
Changes incorporated!